src/layers/core/point-density-hexagon-layer/point-density-hexagon-layer.js

```diff
@@ -64,14 +66,38 @@ export default class PointDensityHexagonLayer extends Layer {
       props.radius = defaultRadius;
     }
 
+    if (Number.isFinite(props.upperPercentile) &&
```

Comment:
Not necessary to fix for this PR but still wanted to comment:

That's a lot of error checking code in the constructor. This makes me feel we really need the `propTypes` feature you created in the example to be added to the library.  For now, you could add your own utility function or file here in this layer, use it in the constructor, and we could build on that.

```
props = checkPropTypes(propTypes, props);
```

Also, we are not supposed to modify props (or any object the caller passes in). `checkPropTypes` could take care of that by cloning when changing (and maintain performance by only cloning when issuing a warning and actually replacing a prop).